docs/site/tutorials/core/5-extension-point-extensions.md

```diff
@@ -0,0 +1,358 @@
+---
+lang: en
+title: 'Extension point and extensions'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/core-tutorial-part5.html
+---
+
+[Extension Point/extension pattern](https://wiki.eclipse.org/FAQ_What_are_extensions_and_extension_points%3F)
+promotes loose coupling and offers great extensibility.
+
+We'll use the following scenario to walk through the important steps to organize
+the `GreetingService` that allows extensible languages - each of them being
+supported by a `Greeter` extension.
+
+![greeters](../../imgs/tutorials/core/greeters.png)
+
+Various constructs from LoopBack 4, such as `Context`, `@inject.*`, and
+`Component` are used to build the service in an extensible fashion.
+
+## Define an extension point
+
+In our scenario, we want to allow other modules to extend or customize how
+people are greeted in different languages. To achieve that, we declare the
+`greeter` extension point, which declares a contract as TypeScript interfaces
+that extensions must conform.
+
+### Define interface for extensions
+
+An extension point interacts with unknown number of extensions. It needs to
+define one or more interfaces as the contracts that each extension must
+implements.
+
+```ts
+/**
+ * Typically an extension point defines an interface as the contract for
+ * extensions to implement
+ */
+export interface Greeter {
+  language: string;
+  greet(name: string): string;
+}
+```
+
+### Define class for the extension point
+
+Typically an extension point is defined as a TypeScript class and bound to a
+context. In our case, we mark `GreetingService` as the extension point that
```

Comment:
I found `we mark 'GreetingService' as the extension point that needs to access a list of greeters` is a bit confusing.

 Do you mean `we mark 'GreetingService' as the extension point that discovers and controls corresponding extensions`? Please correct me if my understanding is incorrect.

Reply:
How about:
```
we mark `GreetingService` as the extension point that has
access to a list of greeters which are defined as extensions.
```